src/Bcgov/DesignSystemPlugin/DocumentRepository/src/Service/DocumentUploader.php

```diff
@@ -0,0 +1,400 @@
+        }
+
+        return $post_id;
+    }
+
+    /**
+     * Get document data including metadata.
+     *
+     * @param int $document_id Document post ID.
+     * @return array Document data.
+     */
+    public function get_document_data( int $document_id ): array {
+        $post = get_post( $document_id );
+
+        if ( ! $post ) {
+            return [];
+        }
+
+        // Get metadata manager.
+        $metadata_manager = new DocumentMetadataManager( $this->config );
+
+        // Return document data.
+        return [
+            'id'       => $post->ID,
+            'title'    => $post->post_title,
+            'date'     => $post->post_date,
+            'author'   => get_the_author_meta( 'display_name', $post->post_author ),
+            'metadata' => $metadata_manager->get_document_metadata( $post->ID ),
+        ];
+    }
+
+    /**
+     * Check for duplicate document title.
+     *
+     * @param string $title Document title to check.
+     * @return \WP_Post|null Duplicate post or null if none found.
+     */
+    private function check_for_duplicate( string $title ) {
+        $query = new \WP_Query(
+            [
+                'post_type'      => $this->config->get_post_type(),
+                'post_status'    => 'publish',
+                'title'          => $title,
+                'posts_per_page' => 1,
+                'fields'         => 'ids',
+            ]
+        );
+
+        if ( $query->have_posts() ) {
+            return get_post( $query->posts[0] );
+        }
+
+        return null;
+    }
+
+    /**
+     * Get human-readable error message for upload error code.
+     *
+     * @param int $error_code PHP upload error code.
+     * @return string Error message.
+     */
+    private function get_upload_error_message( int $error_code ): string {
+        $upload_errors = [
+            UPLOAD_ERR_INI_SIZE   => 'The uploaded file exceeds the upload_max_filesize directive in php.ini.',
+            UPLOAD_ERR_FORM_SIZE  => 'The uploaded file exceeds the MAX_FILE_SIZE directive in the HTML form.',
+            UPLOAD_ERR_PARTIAL    => 'The uploaded file was only partially uploaded.',
+            UPLOAD_ERR_NO_FILE    => 'No file was uploaded.',
+            UPLOAD_ERR_NO_TMP_DIR => 'Missing a temporary folder.',
+            UPLOAD_ERR_CANT_WRITE => 'Failed to write file to disk.',
+            UPLOAD_ERR_EXTENSION  => 'A PHP extension stopped the file upload.',
+        ];
+
+        return $upload_errors[ $error_code ] ?? 'Unknown upload error.';
+    }
+
+    /**
+     * Delete a document and its attachment.
+     *
+     * @param int $document_id Document post ID.
+     * @return bool Whether the deletion was successful.
+     */
+    public function delete_document( int $document_id ): bool {
+        // Get attachment ID.
+        $attachment_id = get_post_meta( $document_id, 'document_file_id', true );
+
+        // Delete document post.
```

Comment:
```suggestion
        // Delete document post, all the metadata, and bypasses the Trash folder
```
**Suggestion** maybe include a slightly better comment so its clear what actually gets deleted.